src/core/src/model.cpp

```diff
@@ -220,11 +220,6 @@ void ov::Model::prerequirements(bool detect_variables, bool detect_parameters) {
 void ov::Model::validate_nodes_and_infer_types() const {
```

Comment:
is it possible to add a test with a graph when this check is not needed?

Reply:
I agree, we need to double check if we have these tests or not but not in this PR